compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -5896,20 +5896,41 @@ private BType getTypeOfTypeParameter(BType selectType, Location pos) {
     private BType getQueryMapConstraintType(BType type, Location pos) {
         if (type.tag == TypeTags.ARRAY) {
             BArrayType arrayType = (BArrayType) type;
-            if (arrayType.state != BArrayState.OPEN && arrayType.size == 2 &&
-                    TypeTags.isStringTypeTag(arrayType.eType.tag)) {
+            if (arrayType.state != BArrayState.OPEN && arrayType.size == 2 && isString(arrayType.eType)) {
                 return arrayType.eType;
             }
         } else if (type.tag == TypeTags.TUPLE) {
             List<BType> tupleTypeList = ((BTupleType) type).tupleTypes;
-            if (tupleTypeList.size() == 2 && TypeTags.isStringTypeTag(tupleTypeList.get(0).tag)) {
+            if (tupleTypeList.size() == 2 && isString(tupleTypeList.get(0))) {
```

Comment:
Why can't we just do `org.wso2.ballerinalang.compiler.semantics.analyzer.Types#isAssignable(org.wso2.ballerinalang.compiler.semantics.model.types.BType, org.wso2.ballerinalang.compiler.semantics.model.types.BType)` with `org.wso2.ballerinalang.compiler.semantics.model.SymbolTable#stringType` as the target?

Reply:
+1. Will update. Thanks.

Reply:
Updated in https://github.com/ballerina-platform/ballerina-lang/pull/37402/commits/385f06c9c723b66f489d8f370eb38cb49333e481.